src/kernel/arch/x86/cmos.zig

```diff
@@ -0,0 +1,458 @@
+const std = @import("std");
+const builtin = @import("builtin");
+const is_test = builtin.is_test;
+const expectEqual = std.testing.expectEqual;
+const build_options = @import("build_options");
+const mock_path = build_options.arch_mock_path;
+const arch = if (is_test) @import(mock_path ++ "arch_mock.zig") else @import("arch.zig");
+const log = @import("../../log.zig");
+
+/// The current year to be used for calculating the 4 digit year, as the CMOS return the last two
+/// digits of the year.
+const CURRENT_CENTURY: u8 = 2000;
+
+/// The port address for the CMOS command register.
+const ADDRESS: u16 = 0x70;
+
+/// The port address for the CMOS data register.
+const DATA: u16 = 0x71;
+
+/// The register location for returning the seconds, (0 - 59).
+const REGISTER_SECOND: u8 = 0x00;
+
+/// The register location for returning the minute, (0 - 59).
+const REGISTER_MINUTE: u8 = 0x02;
+
+/// The register location for returning the hours, (0 - 23 or 0 - 12 depending if a 12hr or 24hr
+/// clock).
+const REGISTER_HOUR: u8 = 0x04;
+
+/// The register location for returning the weekday, (0 - 6). Very unreliable, so will calculate
+/// the day of the week instead.
+const REGISTER_WEEKDAY: u8 = 0x06;
+
+/// The register location for returning the day, (0 - 31).
+const REGISTER_DAY: u8 = 0x07;
+
+/// The register location for returning the month, (0 - 11).
+const REGISTER_MONTH: u8 = 0x08;
+
+/// The register location for returning the year, (0 - 99).
+const REGISTER_YEAR: u8 = 0x09;
+
+/// The register location for returning the century.
+const REGISTER_CENTURY: u8 = 0x32;
```

Comment:
Like the status register variables, these could also be returned from a function in RtcRegister.